### PR TITLE
feat: Genesis settings plugin and Simplified project structure

### PR DESCRIPTION
**Pull request by khouari1**
Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
No

Have you checked all new or changed links?
Yes

Is there anything else you would like us to know?
No

For reference: 

  - We have an [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4)
  - We have a [style guide](https://www.notion.so/genesisglobal/Documentation-style-guide-5b04ec6fe12f4262b90d192effd8059b) 

**This week's exciting excerpts from the style guide**

- We write in UK English, with Oxford English Dictionary (OED) spellings, grammar and vocabulary.  

Use the present tense wherever possible. Only use another tense where it is strictly necessary.

- Present tense (preferred whenever possible): *The script creates a new folder.*
- Future tense (avoid whenever possible): *The script will create a new folder.*

### CODE REVIEW

**Review by AliAlbaity — COMMENTED** [review]
thoughts on having some docs around:
1. localDaoGenVersion property
2. configuration cache even if it's brief

File: docs/05_operations/13_project-structure/01_introduction.mdx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+sidebar_position: 1
+id: introduction
+keywords: [operations, project structure, introduction]
+tags:
+    - operations
+    - project structure
+    - introduction
+---
+
+This section will explain the Genesis Gradle Settings plugin and the Simplified project structure. The non-simplified project structure is referred to as the Legacy structure.
```
Comment (kievitsp) [inline docs/05_operations/13_project-structure/01_introduction.mdx:13]: I think we could do with more of an explanation here. We should introduce the legacy and simplified structure here. We should explain the benefits of the simplified structure and the reasons for a developer to move to that structure.

**Review by kievitsp — COMMENTED** [review]
This looks good. There is a lot of information here. I made a few comments regarding wording, but it would be good to maybe rearrange the structure here, and really focus on what we're trying to tell here to get the order right. 

For example; imagine an experienced genesis developer approaching this. Before we can discuss the settings plugin, perhaps we should discuss the differences between the simple and legacy structure. Also a new developer, they might want to know about the different structures before we get into the settings plugin. 

Also, the other point would be more of a focuss on what it means for these developers. What does a simplified project give compared to a legacy structure? When would they choose one or the other? What benefit does the settings plugin give a developer? Those are the main questions we should address at the start of these pages. 

The benefits really are: 
* centralised configuration
* massive reduction in the number of lines of code in your gradle files
* consistent configuration, i.e. you won't need to remember to add a dictionary dependency
* simplified development experience in the intellij plugin

File: docs/05_operations/13_project-structure/03_simplified-structure.mdx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+You are not limited to putting everything in this one module, you can add as many as you like but there is no longer a requirement to have more than one.
+
+By default, the main module will be detected by a module ending with `-app`. There is a property in the Settings plugin to specify a custom name which you can read about [here](../settings-plugin/#main-module-name).
+
+Moving forward, this is the recommended structure for projects and is a pre-requisite for PBC development.
+
+## How to create a project
+
+To create a new project with the Simplified structure, you create your project as usual using genx with the blank-app-seed but also specify the following option `--ref simple-app-seed`.
```
[the remark this answers is not in the document]
Comment (khouari1) [reply docs/05_operations/13_project-structure/03_simplified-structure.mdx:26]: Yep, fair point.

**Comment by khouari1** [comment]
> thoughts on having some docs around:
> 
> 1. localDaoGenVersion property
> 2. configuration cache even if it's brief

Yes, I think both of those things would be useful. As not directly related to this, they can be done in a separate change.

File: docs/05_operations/13_project-structure/02_settings-plugin.mdx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    projectType = APPLICATION
+}
+```
+
+### Dependencies
+
+Here you should declare dependencies on other Genesis products, then the Settings plugin can take care of where they should be added automatically. 
+The Settings plugin will do the following with these dependency declarations:
+- Add a dependency on the product's config to your project's dictionary cache for code generation
+- Capture runtime details of the product's processes (e.g. for running processes with the IntelliJ plugin or the Deploy plugin)
```
[the remark this answers is not in the document]
Comment (khouari1) [reply docs/05_operations/13_project-structure/02_settings-plugin.mdx:99]: 1. Fair enough
2. Does it do that? I don't recall seeing that but may have forgotten.

[the remark this answers is not in the document]
Comment (kievitsp) [reply docs/05_operations/13_project-structure/02_settings-plugin.mdx:99]: yes it does

File: docs/05_operations/13_project-structure/02_settings-plugin.mdx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+There are a few common configuration options to both Legacy and Simplified listed below.
+
+### Project type
+
+There are two project types: PBC (Packaged Business Capability) and APPLICATION. PBC should be set when the project contains reusable code and functionality that will be imported by other projects, otherwise APPLICATION should be used (and is the default).
+
+Differences between the two:
+- When the project type is PBC, by default, the project's dictionary cache sub-modules have publishing tasks disabled. If you want to enable publishing then disable the `genesisPublicationDisabler` plugin (see [here](#plugins) for details).
+- When the project type is APPLICATION, a site-specific distribution is created in addition to the project's one.
```
[the remark this answers is not in the document]
Comment (khouari1) [reply docs/05_operations/13_project-structure/02_settings-plugin.mdx:86]: Yes, I will make that more explicit.

File: docs/05_operations/13_project-structure/02_settings-plugin.mdx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+</TabItem>
+<TabItem value="Legacy Structure">
+
+| Plugin Name | Plugin ID | Description | Enabled by default |
+| --- | --- | --- | --- |
+| Kotlin | org.jetbrains.kotlin.jvm | Kotlin Gradle plugin | Yes |
+| Genesis Build | global.genesis.build | Core Genesis plugin to enable build tasks, including code generation | Yes |
+| Genesis Distribution | global.genesis.distribution | Creates the project's server distribution zip file. At the moment it does this based on default configuration within the plugin. If you want to customise your distribution then you will need to disable this plugin and create your own configuration using the Gradle distribution plugin. In the future we will support customisation from this plugin directly. | Yes |
+| Genesis Exec | global.genesis.exec | Creates Gradle tasks used by the Genesis IntelliJ plugin | Yes |
```
Comment (ShrutiTalewad) [inline docs/05_operations/13_project-structure/02_settings-plugin.mdx:138]: I am not sure whether to mention about exec plugin yet! as all its functionality is exposed through intellij plugin

File: docs/05_operations/13_project-structure/02_settings-plugin.mdx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+<Tabs defaultValue="Simplified Structure" values={[{ label: 'Simplified Structure', value: 'Simplified Structure', },{ label: 'Legacy Structure', value: 'Legacy Structure'}]}>
+<TabItem value="Simplified Structure">
+
+| Plugin Name | Plugin ID | Description | Enabled by default |
+| --- | --- | --- | --- |
+| Kotlin | org.jetbrains.kotlin.jvm | Kotlin Gradle plugin | Yes |
+| Genesis Build | global.genesis.build | Core Genesis plugin to enable build tasks, including code generation | Yes |
+| Genesis Distribution | global.genesis.distribution | Creates the project's server distribution zip file. At the moment it does this based on default configuration within the plugin. If you want to customise your distribution then you will need to disable this plugin and create your own configuration using the Gradle distribution plugin. In the future we will support customisation from this plugin directly. | Yes |
+| Genesis Publication Disabler | global.genesis.publication-disabler | Used to disable dictionary-cache sub-module publishing tasks if the project is of type PBC | Yes |
+| Genesis Exec | global.genesis.exec | Creates Gradle tasks used by the Genesis IntelliJ plugin | Yes |
```
Comment (ShrutiTalewad) [inline docs/05_operations/13_project-structure/02_settings-plugin.mdx:127]: I am not sure whether to mention about exec plugin yet! as all its functionality is exposed through intellij plugin

File: docs/05_operations/13_project-structure/02_settings-plugin.mdx (hunk 5 above)
Comment (khouari1) [reply docs/05_operations/13_project-structure/02_settings-plugin.mdx]: Hmm fair point, but as it is exposed via the plugins block in the Settings plugin, I think we need to mention it for consistency. We could hide it from the plugins block as an alternative.

File: docs/05_operations/13_project-structure/03_settings-plugin.mdx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+```kotlin
+genesis {
+    projectType = APPLICATION
+}
+```
+
+### Dependencies
+
+Here you should declare dependencies on other Genesis products, then the Settings plugin can take care of where they should be added automatically.
```
Comment (khouari1) [inline docs/05_operations/13_project-structure/03_settings-plugin.mdx:101]: Note to self: Link to PBCs

**Review by wjhendry — APPROVED** [review]

**Pull request merged**